### PR TITLE
introduction freertos

### PR DESCRIPTION
**Pull request by arafaarafa**

### CODE REVIEW

File: S_RTOS_01IntroductionToFreertos/Task_2/main.c [hunk 1]
```diff
@@ -88,22 +88,71 @@ static void prvSetupHardware( void );
...
 /*
  * Application entry point:
  * Starts all the other tasks, then starts the scheduler. 
  */
 int main( void )
 {
+	/*global vars*/
+
+	task_1_cfg.pin_num = PIN1;
+	task_1_cfg.delay = 100;
```
Comment (ahmedatef23) [inline S_RTOS_01IntroductionToFreertos/Task_2/main.c:110]: don't use the magic number

File: S_RTOS_01IntroductionToFreertos/Task_3/button_task/button_task.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		if(button_state == PIN_IS_HIGH){
+			
+			while(button_state == PIN_IS_HIGH){
+				vTaskDelay(1000);
+				sec_counter++;
+				button_state = GPIO_read(PORT_0, ((button_task_config *)pvParameters)->pin_num);
+			}
+			if(sec_counter>4){
+				*(((button_task_config *)pvParameters)->led_delay)= 100;
+			}else if((sec_counter>2) && (sec_counter<4)){
```
Comment (ahmedatef23) [inline S_RTOS_01IntroductionToFreertos/Task_3/button_task/button_task.c:21]: don't use magic number

Comment (arafaarafa) [reply S_RTOS_01IntroductionToFreertos/Task_3/button_task/button_task.c:21]: ok, thanks for your interest, I will resolve it.

File: S_RTOS_01IntroductionToFreertos/Task_2/main.c (hunk 1 above)
Comment (arafaarafa) [reply S_RTOS_01IntroductionToFreertos/Task_2/main.c:110]: https://github.com/arafaarafa/Sprints_Tasks/pull/10/files
all magic numbers deleted

**Pull request closed without merging**